src/installer/tests/HostActivation.Tests/DependencyResolution/PerAssemblyVersionResolutionMultipleFrameworks.cs

```diff
@@ -0,0 +1,200 @@
+        private const string TestAssemblyWithFileVersion = "Test.Assembly.FileVersion";
+        private const string TestAssemblyWithBothVersions = "Test.Assembly.BothVersions";
+
+        [Theory]
+        [InlineData(TestAssemblyWithBothVersions, null, null, MicrosoftNETCoreApp)] // NetCoreApp has higher version than HighWare
+        [InlineData(TestAssemblyWithBothVersions, "1.0.0.0", "1.0.0.0", MicrosoftNETCoreApp)]
+        [InlineData(TestAssemblyWithBothVersions, "3.0.0.0", "4.0.0.0", null)]  // App has higher version than any framework
+        [InlineData(TestAssemblyWithBothVersions, "2.1.1.1", "3.3.0.0", null)]  // App has higher file version
+        [InlineData(TestAssemblyWithBothVersions, "2.1.1.1", "3.2.2.2", MicrosoftNETCoreApp)]  // Lower level framework always wins on equality (this is intentional)
+        [InlineData(TestAssemblyWithBothVersions, null, "4.0.0.0", MicrosoftNETCoreApp)] // The one with version wins
+        [InlineData(TestAssemblyWithBothVersions, null, "2.0.0.0", MicrosoftNETCoreApp)] // The one with version wins
+        [InlineData(TestAssemblyWithBothVersions, "3.0.0.0", null, null)]
+        [InlineData(TestAssemblyWithBothVersions, "2.1.1.1", null, MicrosoftNETCoreApp)]
+        [InlineData(TestAssemblyWithNoVersions, null, null, MicrosoftNETCoreApp)] // No versions are treated as equal (so lower one wins)
+        [InlineData(TestAssemblyWithNoVersions, "1.0.0.0", null, null)] // The one with version wins
+        [InlineData(TestAssemblyWithNoVersions, "1.0.0.0", "1.0.0.0", null)] // The one with version wins
+        [InlineData(TestAssemblyWithNoVersions, null, "1.0.0.0", null)] // The one with version wins
+        [InlineData(TestAssemblyWithAssemblyVersion, null, null, HighWare)] // Highware has higher version than NetCoreApp
+        [InlineData(TestAssemblyWithAssemblyVersion, "1.0.0.0", null, HighWare)]
+        [InlineData(TestAssemblyWithAssemblyVersion, null, "1.0.0.0", HighWare)]
+        [InlineData(TestAssemblyWithAssemblyVersion, "3.0.0.0", "1.0.0.0", null)] // App has higher version than any framework
+        [InlineData(TestAssemblyWithAssemblyVersion, "2.1.1.2", null, HighWare)] // Both are exactly the same, so lower level wins
+        [InlineData(TestAssemblyWithAssemblyVersion, "2.1.1.2", "1.0.0.0", null)]
+        [InlineData(TestAssemblyWithFileVersion, null, null, MicrosoftNETCoreApp)] // Frameworks both have the same version - lower one wins
+        [InlineData(TestAssemblyWithFileVersion, "1.0.0.0", null, null)] // App has assembly version, no framework has it - so app wins
+        [InlineData(TestAssemblyWithFileVersion, null, "1.0.0.0", MicrosoftNETCoreApp)]
+        [InlineData(TestAssemblyWithFileVersion, null, "4.0.0.0", null)] // App has higher version than either framework
+        [InlineData(TestAssemblyWithFileVersion, null, "3.2.2.2", MicrosoftNETCoreApp)] // Exactly equal - lower one wins
```

Comment:
I feel like I've asked this before, but what does `lower` mean here?

Reply:
Oh, this is the dependency hierarchy / depth thing, right? So app is highest level and NETCoreApp would always be lowest?

Reply:
Yes - the app is the highest level, NETCoreApp is always the "root" lowest level. In the resolutio logic, if everything else is equal the lowest level typically wins.